### DISCUSSION TITLE
built-in output archive encryption

### DISCUSSION THREAD

**Discussion by sirbrowser** [tclahr/uac#148] [Ideas]
Hello, 

I wonder if it was possible to add an option to generate an encrypted archive.

Thanks for your work on this amazing tool!

**Comment by tclahr** [comment]
Hi,
Unfortunately, I don't see too many possibilities for that. UAC relies on and uses built-in tools provided by the target OS, so the target OS should have a tool for encrypting or generating an encrypted archive.

One possibility would be using zip to create a password-protected archive. But, zip is not available on some systems, and even if it is, some systems do not provide a version that can create a password-protected archive. UAC could provide an option to create an encrypted archive if zip is available in the target OS and if it supports the creation of password-protected archives.

Do you have anything in mind? Any other tool that could be used for accomplishing that?

Thanks!

**Comment by sirbrowser** [comment]
Hello,

I agree that is what makes uac powerful to rely on built-in tools.

I may have two possible solutions in mind : 
- I think GPG could be of interest for encrypting, I assume the package is installed by default on multiple distributions but I may be wrong.
- An other one could be to ship uac with the "zip" package as you do for stat.pl for example in case stat or statx is not available on the targeted system -> this option could be even more interesting as in addition to add encryption we could specify the output desired format instead of the default ".tar.gz" format.

Please let me know what do you think, thanks!

**Comment by tclahr** [comment]
I think zip would be the best option since it is available on most systems. I will convert this into an issue so it can be tracked.
